src/main/java/net/starschema/clouddb/cmdlineverification/Oauth2Bigquery.java

```diff
@@ -265,11 +260,13 @@ public static Bigquery authorizeviaservice(String serviceaccountemail,
                                                String keypath,
                                                String password,
                                                String userAgent,
-                                               Integer connectTimeout,
-                                               Integer readTimeout) throws GeneralSecurityException, IOException {
+                                               String jsonAuthContents, Integer readTimeout, Integer connectTimeout) throws GeneralSecurityException, IOException {
         GoogleCredential credential;
         // Determine which keyfile we are trying to authenticate with.
-        if (Pattern.matches(".*\\.json$", keypath)) {
+        if (jsonAuthContents != null) {
+            credential = Oauth2Bigquery.createJsonCredential(jsonAuthContents);
+        } else if (Pattern.matches(".*\\.json$", keypath)) {
+            // For backwards compat: this is no longer the preferred path for JSON (better to use [jsonAuthContents]
```

Comment:
Makes sense to keep the backward-compatible `keypath` codepath around, but won't the next line fail? `createJsonCredential`, if passed a keypath instead of a string representing auth contents, will no longer `File jsonKey = new File(keypath);` before trying to build its inputstream, right? 

Maybe I'm missing some Java magic that will look for a file in that keypath before trying to build an inputstream from it